lib/std/Build.zig

```diff
@@ -1286,11 +1286,14 @@ pub fn step(b: *Build, name: []const u8, description: []const u8) *Step {
 }
 
 pub const StandardOptimizeOptionOptions = struct {
-    preferred_optimize_mode: ?std.builtin.OptimizeMode = null,
+    /// When set, this option causes the standard optimize option to change to a simple boolean flag, `-Drelease`,
+    /// choosing between Debug mode and the mode given here.
+    /// Note that this does not change the default optimization mode.
+    preferred_release_mode: ?std.builtin.OptimizeMode = null,
 };
 
 pub fn standardOptimizeOption(b: *Build, options: StandardOptimizeOptionOptions) std.builtin.OptimizeMode {
```

Comment:
Maybe panic early here if `preferred_release_mode == .Debug`?

Reply:
hmmm, it definitely wouldn't make much sense for someone to do that but I'm not sure I see too much reason to disallow it? I have no strong opinion either way

Reply:
yeah that seems arbitrary for something unlikely to actualize

Reply:
We're specifically renaming this field from `preferred_optimize_mode` (a name that associates it with the `-O` flag, which allows release modes *and* `Debug`) to `preferred_release_mode`, a name that supposedly better suits its intended purpose. But since it's typed as `?std.builtin.OptimizeMode`, it still allows `Debug`. That means we're trading one kind of confusion for another.

Additionally, given that this is a 'standard option' that's supposed to represent Zig good practice, I think we can very reasonably take the opinionated stance that nobody should make their build script support `-Drelease` and have it mean a debug build. Or if they want to, they have to go out of their way.

In other words: Invalid/nonsensical states shouldn't be representable. Using an inline `enum { ReleaseFast, ReleaseSmall, ReleaseSafe }` and translating to `OptimizeMode` on return would be the 'right' way to do that, but I acknowledge that it might be useful to be able to pass in an existing `OptimizeMode` value that's on hand, hence the panic suggestion.